### PR TITLE
Add batchSize option to uploader plugin

### PR DESCRIPTION
**Pull request by PeterChaplin**
Testing
----
- Used js-concat to have mla-www use this version on DEV
- Used https://www.dev.mlaccessories.co.uk/admin/catalogue-tile to upload images
- Modified `jquery.uploaderUI.js` to use `batchSize: 3`
- Uploaded a 86.4 MB pdf
- Used network panel to confirm only 3 requests were made at a time

### CODE REVIEW

**Comment by PeterChaplin** [comment]
Updated and re-tested as above, with and without batch size specified.

File: js/qcode.uploader.js [hunk 1]
```diff
@@ -86,41 +74,183 @@ qcode.Uploader = (function() {
...
+
+            // Start first batch
+            batches[0].start(
+                uploader.url,
+                uploader.headers,
+                uploader.postData
+            );
+        },
+        setChunkProgress: function(index, progress) {
+            // Set the numer of bytes uploaded
```
Comment (daniel-qcode) [inline js/qcode.uploader.js]: typo `number`

File: js/qcode.uploader.js [hunk 2]
```diff
@@ -86,41 +74,183 @@ qcode.Uploader = (function() {
...
+            });
+
+            // Start first batch
+            batches[0].start(
+                uploader.url,
+                uploader.headers,
+                uploader.postData
+            );
+        },
+        setChunkProgress: function(index, progress) {
```
Comment (daniel-qcode) [inline js/qcode.uploader.js]: could we change `index` -> `chunkIndex`

Comment (daniel-qcode) [reply js/qcode.uploader.js]: could we change `progress` -> `bytes`

File: js/qcode.uploader.js [hunk 3]
```diff
@@ -86,41 +74,183 @@ qcode.Uploader = (function() {
...
+                    start = end;
+                    end = start + chunkSize;
+                }
+            } else {
+                chunks.push(file);
+            }
+            return chunks;
+        },
+        _createBatches: function(chunks, batchSize) {
+            // Create request batch objects to upload the chunks, begin upload
```
Comment (daniel-qcode) [inline js/qcode.uploader.js]: can you rewrite this comment as this function is not responsible for uploading

**Comment by daniel-qcode** [comment]
Are you able to confirm testing for error event? is outstanding requests in current batch cancelled and no further batches sent?

Are you able to confirm testing for progress event? does progress get reported correctly when there is 1 chunk required to upload? does progress get reported correctly when there are multiple chunks required to upload

**Comment by daniel-qcode** [comment]
could you "make" new qcode-ui release so that it is available on js.qcode.co.uk please.

**Comment by PeterChaplin** [comment]
Release 4.29.0 made

**Pull request merged**